guides/keras_hub/classification_with_keras_hub.py

```diff
@@ -0,0 +1,616 @@
+this case to avoid adding noise to our evaluation metric due to applying random
+augmentations.
+"""
+
+inference_resizing = keras.layers.Resizing(*IMAGE_SIZE, crop_to_aspect_ratio=True)
+
+
+def do_resize(inputs):
+    inputs["images"] = inference_resizing(inputs["images"])
+    return inputs
+
+
+eval_ds = eval_ds.map(do_resize, num_parallel_calls=tf.data.AUTOTUNE)
+
+image_batch = next(iter(eval_ds.take(1)))["images"]
+plot_image_gallery(
+    image_batch,
+)
+
+"""
+Finally, lets unpackage our datasets and prepare to pass them to `model.fit()`,
+which accepts a tuple of `(images, labels)`.
+"""
+
+
+def unpackage_dict(inputs):
+    return inputs["images"], inputs["labels"]
+
+
+train_ds = train_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Data augmentation is by far the hardest piece of training a modern
+classifier.
+Congratulations on making it this far!
+
+### Optimizer Tuning
+
+To achieve optimal performance, we need to use a learning rate schedule instead
+of a single learning rate. While we won't go into detail on the Cosine decay
+with warmup schedule used here,
+[you can read more about it here](https://scorrea92.medium.com/cosine-learning-rate-decay-e8b50aa455b).
+"""
+
+
+def lr_warmup_cosine_decay(
+    global_step,
+    warmup_steps,
+    hold=0,
+    total_steps=0,
+    start_lr=0.0,
+    target_lr=1e-2,
+):
+    # Cosine decay
+    learning_rate = (
+        0.5
+        * target_lr
+        * (
+            1
+            + ops.cos(
+                math.pi
+                * ops.convert_to_tensor(
+                    global_step - warmup_steps - hold, dtype="float32"
+                )
+                / ops.convert_to_tensor(
+                    total_steps - warmup_steps - hold, dtype="float32"
+                )
+            )
+        )
+    )
+
+    warmup_lr = target_lr * (global_step / warmup_steps)
+
+    if hold > 0:
+        learning_rate = ops.where(
+            global_step > warmup_steps + hold, learning_rate, target_lr
+        )
+
+    learning_rate = ops.where(global_step < warmup_steps, warmup_lr, learning_rate)
+    return learning_rate
+
+
+class WarmUpCosineDecay(schedules.LearningRateSchedule):
+    def __init__(self, warmup_steps, total_steps, hold, start_lr=0.0, target_lr=1e-2):
+        super().__init__()
+        self.start_lr = start_lr
+        self.target_lr = target_lr
+        self.warmup_steps = warmup_steps
+        self.total_steps = total_steps
+        self.hold = hold
+
+    def __call__(self, step):
+        lr = lr_warmup_cosine_decay(
+            global_step=step,
+            total_steps=self.total_steps,
+            warmup_steps=self.warmup_steps,
+            start_lr=self.start_lr,
+            target_lr=self.target_lr,
+            hold=self.hold,
+        )
+
+        return ops.where(step > self.total_steps, 0.0, lr)
```

Comment:
Why set the LR to 0? Should it not be a nonzero min_lr value?

Reply:
no, with cosine decay the lr decays to 0 at the end of training. Actually, we wouldn't need this conditional anyway, since training can't get past `total_steps`. I assume this was mainly for plotting purposes (it's been copied from the old version of this guide).